clients/java/zts/src/test/java/com/yahoo/athenz/zts/ZTSClientTest.java

```diff
@@ -4065,4 +4065,29 @@ public void testPostExternalCredentialsRequest() throws IOException, URISyntaxEx
             assertEquals(400, ex.getCode());
         }
     }
+
+    @Test
+    public void testCanSetProxyConfiguration() {
+        ServiceIdentityProvider siaProvider = Mockito.mock(ServiceIdentityProvider.class);
+        ZTSClient client = new ZTSClient("http://localhost:4080/",
+                "http://localhost:8080", "iaas.athenz", "ci", siaProvider);
+        client.close();
+    }
+
+    @Test
+    public void testInvalidProxyException() {
+        ServiceIdentityProvider siaProvider = Mockito.mock(ServiceIdentityProvider.class);
+
+        ZTSClient client = null;
+        try {
+            client = new ZTSClient("http://localhost:4080/",
+                    "invalid-proxy", "iaas.athenz", "ci", siaProvider);
+            fail();
```

Comment:
Can you please add test cases where

- domainName
- serviceName
- siaProvider

are null and IllegalArgumentException is thrown.

Reply:
Thank you for the feedback. I have added the tests.